### PR TITLE
Add commands to collect branch coverage

### PR DESCRIPTION
**Pull request by Navidem**
Generates branch coverage along with other text coverage info, and copies the collected branch profiles to be used by fuzz introspector.

### CODE REVIEW

File: infra/base-images/base-runner/coverage [hunk 1]
```diff
@@ -55,6 +55,9 @@ PATH_EQUIVALENCE_ARGS="-path-equivalence=/,$OUT"
 LLVM_COV_COMMON_ARGS="$PATH_EQUIVALENCE_ARGS \
     -ignore-filename-regex=.*src/libfuzzer/.* $COVERAGE_EXTRA_ARGS"
 
+# Options to extract branch coverage.
+BRANCH_COV_ARGS="--show-branches=count --show-expansions"
```
Comment (oliverchang) [inline infra/base-images/base-runner/coverage:59]: Does this significantly increase size of the resulting reports?

**Review by oliverchang — APPROVED** [review]
LGTM. this is blocked on the introspector side changes landing right?

File: infra/base-images/base-runner/coverage (hunk 1 above)
Comment (Navidem) [reply infra/base-images/base-runner/coverage:59]: It definitely increases the report size as it is exposing coverage information for each branch in the code. The amount of increase is project dependent, for the sample project (`htslib`) that I checked, the increase is 2X.

**Comment by Navidem** [comment]
> LGTM. this is blocked on the introspector side changes landing right?

Well, these two PRs can land independently and things (except branch blocker) should work if the other is not yet merged.

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer introspector --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
Uploaded https://github.com/google/oss-fuzz/pull/7727 to fix the issue with the trial builds not triggering.

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer introspector --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer afl --fuzzing-engine libfuzzer

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer address --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer introspector --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
/gcbrun trial_build.py libarchive opencv --sanitizer coverage --fuzzing-engine libfuzzer

**Comment by oliverchang** [comment]
let's not block this PR any more and get it merged.

**Pull request merged**